crates/oxc_semantic/src/module_record/builder.rs

```diff
@@ -85,15 +85,14 @@ impl ModuleRecordBuilder {
         for ee in export_entries {
             // a. If ee.[[ModuleRequest]] is null, then
             if ee.module_request.is_none() {
-                let local_name = match &ee.local_name {
-                    ExportLocalName::Name(name) => Some(name),
+                let found_import_entry = match &ee.local_name {
+                    ExportLocalName::Name(name) => self
+                        .module_record
+                        .import_entries
+                        .iter()
+                        .find(|entry| entry.local_name.name() == name.name()),
                     _ => None,
                 };
-                let found_import_entry = self
-                    .module_record
-                    .import_entries
-                    .iter()
-                    .find(|import_entry| Some(&import_entry.local_name) == local_name);
                 match found_import_entry {
```

Comment:
I think the performance improvement comes from here. ImportEntry is only looked for if `ee.local_name` is `ExportLocalName::Name`